### ISSUE TITLE
stylelint-disable segments break autofixing for the whole file, even for code outside of the ignored segment

### ISSUE THREAD

**Issue by ArianeBouchardConformit** [hudochenkov/stylelint-order#203]
After using `/* stylelint-disable rule-name */` to disable this plugin's rules, and then re-enabling them, violations are still flagged, but autofixing breaks.

To show a simplified version of the problem, I've reduced my config to only contain `order/properties-order` and `length-zero-no-unit`.

```javascript
            'length-zero-no-unit': [
               true,
               {
                  ignorePreludeOfAtRules: [ 'function', 'mixin' ],
                  severity: 'warning',
               },
            ],
            'order/properties-order': [
               [
                  {
                     emptyLineBefore: 'threshold',
                     properties: [ 'width', 'height' ],
                  },
               ],
               {
                  emptyLineBeforeUnspecified: 'threshold',
                  emptyLineMinimumPropertyThreshold: 5,
                  severity: 'warning',
                  unspecified: 'ignore',
               },
            ],
```

Here's a style snippet to reproduce the issue:

```scss
.filter-special {
   .filter-item {
      height: 0;
      width: 10px;
   }
}

/* stylelint-disable order/order, order/properties-order, length-zero-no-unit */

@mixin button-loading-styles($parent-includes-border: true) {
   position: relative;
}

/* stylelint-enable order/order, order/properties-order, length-zero-no-unit */

.filter-base{
   .filter-item {
      height: 0;
      width: 10px;
   }
}
```

The above contains violations of both rules for `.filter-special` and `.filter-base`:

- `length-zero-no-unit` (0px should be 0)
- `order/properties-order` (width should be before height)

Both violations are reported correctly. But autofix is another story.

If I try to run an autofix, `0px` will get changed to `0`, but the property order will not change.

The issue disappears if I remove both `stylelint-disable`/`stylelint-enable` comments. Then autofix is able to fix `order/properties-order` issues as well.

**Comment by hudochenkov** [comment]
I think this is due change in stylelint 16.8.2: https://stylelint.io/user-guide/options#fix:~:text=When%20a%20rule%20relies%20on%20the%20deprecated%20context.fix%20and%20a%20source%20contains%3A

`stylelint-order` does use `context.fix`.

**Comment by ArianeBouchardConformit** [comment]
I see. Would it be complicated to update to the new, non-deprecated way of doing things?

**Comment by hudochenkov** [comment]
I need to try and see. So far I don't see a quick fix.
